.github/workflows/deploy.yaml

```diff
@@ -58,10 +63,31 @@ jobs:
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
 
       - name: Build and push
+        # On pushes to `main`, we build and push a new image, so we can simply
+        # use the `docker/build-push-action` action.
+        if: ${{ github.event_name == 'push' }}
         uses: docker/build-push-action@v6
         with:
-          push: ${{ github.event_name != 'pull_request' }}
+          push: ${{ github.event_name == 'push' }}
           sbom: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           context: .
+
+      - name: Tag and push
+        # For releases, we specifically do _not_ want to rebuild, just tag the
+        # existing image and push. There's no officially maintained action for
+        # this use case, but it's trivial enough to do ourselves.
+        if: ${{ github.event_name == 'release' }}
+        env:
+          # Tags come in the form of a fully qualified image name and tag, eg:
+          # mozilla/autograph:1.1.8
+          # us-west2-docker.pkg.dev/autograph-proj/autograph-repo/autograph:1.1.8
+          TAGS: ${{ steps.meta.outputs.tags }}
+          SRC: ${{ vars.DOCKERHUB_REPO}}:sha-${{ github.sha }}
+        run: |
+          docker pull $SRC
+          for tag in $TAGS; do
+            docker tag $SRC $tag
```

Comment:
`&&` here? Or have I forgotten that GHA runs with the equivalent of `set -e` in place?

Reply:
AFAICT it does. eg: https://github.com/bhearsum/autograph/actions/runs/11545032840/job/32165991556
```
Run docker pull $SRC
  docker pull $SRC
  for tag in $TAGS; do
    docker tag $SRC $tag
    docker push $tag
  done
  shell: /usr/bin/bash -e {0}
  env:
    DOCKER_METADATA_OUTPUT_VERSION: 1.1.7
    DOCKER_METADATA_OUTPUT_TAGS: mozilla/autograph:1.1.7
  mozilla/autograph:1.1.7-prod
  northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7
  northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7-prod
    DOCKER_METADATA_OUTPUT_LABELS: org.opencontainers.image.created=2024-10-28T15:14:54.108Z
  org.opencontainers.image.description=Mozilla's digital signature service
  org.opencontainers.image.licenses=MPL-2.0
  org.opencontainers.image.revision=e00127f170a4bef2157734245d6fd4efb2f15d39
  org.opencontainers.image.source=https://github.com/bhearsum/autograph
  org.opencontainers.image.title=autograph
  org.opencontainers.image.url=https://github.com/bhearsum/autograph
  org.opencontainers.image.version=1.1.7
    DOCKER_METADATA_OUTPUT_ANNOTATIONS: manifest:org.opencontainers.image.created=2024-10-28T15:14:54.108Z
  manifest:org.opencontainers.image.description=Mozilla's digital signature service
  manifest:org.opencontainers.image.licenses=MPL-2.0
  manifest:org.opencontainers.image.revision=e00127f170a4bef2157734245d6fd4efb2f15d39
  manifest:org.opencontainers.image.source=https://github.com/bhearsum/autograph
  manifest:org.opencontainers.image.title=autograph
  manifest:org.opencontainers.image.url=https://github.com/bhearsum/autograph
  manifest:org.opencontainers.image.version=1.1.7
    DOCKER_METADATA_OUTPUT_JSON: {"tags":["mozilla/autograph:1.1.7","mozilla/autograph:1.1.7-prod","northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7","northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7-prod"],"labels":{"org.opencontainers.image.created":"2024-10-28T15:14:54.108Z","org.opencontainers.image.description":"Mozilla's digital signature service","org.opencontainers.image.licenses":"MPL-2.0","org.opencontainers.image.revision":"e00127f170a4bef2157734245d6fd4efb2f15d39","org.opencontainers.image.source":"https://github.com/bhearsum/autograph","org.opencontainers.image.title":"autograph","org.opencontainers.image.url":"https://github.com/bhearsum/autograph","org.opencontainers.image.version":"1.1.7"},"annotations":["manifest:org.opencontainers.image.created=2024-10-28T15:14:54.108Z","manifest:org.opencontainers.image.description=Mozilla's digital signature service","manifest:org.opencontainers.image.licenses=MPL-2.0","manifest:org.opencontainers.image.revision=e00127f170a4bef2157734245d6fd4efb2f15d39","manifest:org.opencontainers.image.source=https://github.com/bhearsum/autograph","manifest:org.opencontainers.image.title=autograph","manifest:org.opencontainers.image.url=https://github.com/bhearsum/autograph","manifest:org.opencontainers.image.version=1.1.7"]}
    DOCKER_METADATA_OUTPUT_BAKE_FILE_TAGS: /home/runner/work/_temp/docker-actions-toolkit-2xEG3v/docker-metadata-action-bake-tags.json
    DOCKER_METADATA_OUTPUT_BAKE_FILE_LABELS: /home/runner/work/_temp/docker-actions-toolkit-2xEG3v/docker-metadata-action-bake-labels.json
    DOCKER_METADATA_OUTPUT_BAKE_FILE_ANNOTATIONS: /home/runner/work/_temp/docker-actions-toolkit-2xEG3v/docker-metadata-action-bake-annotations.json
    DOCKER_METADATA_OUTPUT_BAKE_FILE: /home/runner/work/_temp/docker-actions-toolkit-2xEG3v/docker-metadata-action-bake.json
    TAGS: mozilla/autograph:1.1.7
  mozilla/autograph:1.1.7-prod
  northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7
  northamerica-northeast2-docker.pkg.dev/bhearsum-test/autographtest/autograph:1.1.7-prod
    SRC: mozilla/autograph:e00127f170a4bef2157734245d6fd4efb2f15d39
Error response from daemon: manifest for mozilla/autograph:e00127f170a4bef2157734245d6fd4efb2f15d39 not found: manifest unknown: manifest unknown
Error: Process completed with exit code 1.
```